ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1088,6 +1088,15 @@ void Matmul::validate(
 
                         TT_FATAL(program_config.out_subblock_w == per_core_N || program_config.out_subblock_h == 1, "Error");
                     }
+                    if (input_tensor_b.buffer()->buffer_type() == tt_metal::BufferType::L1 && input_tensor_b.memory_config().is_sharded()) {
+                        TT_FATAL(this->output_mem_config.memory_layout == TensorMemoryLayout::WIDTH_SHARDED, "Operand B can only be interleaved or L1 width sharded.");
```

Comment:
Is the exception message wrong?

Reply:
No, I believe it's correct. It can be either DRAM/L1 interleaved, or it can be sharded in L1 but then only width sharding is supported.